### ISSUE TITLE
[ITK 5.1|Windows|Static] OpenCL linkage is missing system libraries.

### ISSUE THREAD

**Issue by Neumann-A** [InsightSoftwareConsortium/ITK#1821]
### Description
```
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_Child referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_Device_IDW referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_Device_ID_ListW referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_Device_ID_List_SizeW referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_DevNode_PropertyW referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_DevNode_Status referenced in function ProbeDevice
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Get_Sibling referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Locate_DevNodeW referenced in function khrIcdOsVendorsEnumerateHKR
OpenCL.lib(icd_windows_hkr.c.obj) : error LNK2019: unresolved external symbol __imp_CM_Open_DevNode_Key referenced in function ReadOpenCLKey
lib\_ITKGPUCommonPython.pyd : fatal error LNK1120: 9 unresolved externals
```
changed:
https://github.com/InsightSoftwareConsortium/ITK/blob/a10f19eff8bfff173edb14a2af6f23f33568f2d4/Modules/Core/GPUCommon/CMakeLists.txt#L26-L30

to
```
if(ITK_USE_GPU)
  set(ITKGPUCommon_LIBRARIES ITKGPUCommon cfgmgr32 OneCoreUAP)
  set(ITKGPUCommon_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
  set(ITKGPUCommon_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES} cfgmgr32 OneCoreUAP)
endif()
```
but you probably want to make it Platform independent.

**Comment by seanm** [comment]
Very relatedly, OpenCL linking is failing on macOS here: https://open.cdash.org/viewBuildError.php?buildid=7890136

I "fixed" it with:

```
diff --git a/Modules/Core/GPUCommon/CMakeLists.txt b/Modules/Core/GPUCommon/CMakeLists.txt
index da702aa..1cabeaf 100644
--- a/Modules/Core/GPUCommon/CMakeLists.txt
+++ b/Modules/Core/GPUCommon/CMakeLists.txt
@@ -24,7 +24,7 @@ if(ITK_USE_GPU AND APPLE AND NOT ITK_COMPILER_HAS_BLOCKS)
 endif()
 
 if(ITK_USE_GPU)
-  set(ITKGPUCommon_LIBRARIES ITKGPUCommon)
+  set(ITKGPUCommon_LIBRARIES ITKGPUCommon "-framework OpenCL")
   set(ITKGPUCommon_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
   set(ITKGPUCommon_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
 endif()
```

Maybe someone that knows CMake syntax could help conditionalize my and @Neumann-A's patches to be correctly conditionalized per-OS?

**Comment by dzenanz** [comment]
Fixed by #3437 and #3440.
